### ISSUE TITLE
Fix Localization warnings

### ISSUE THREAD

**Issue by matthiasgeiger** [JabRef/jabref#154]
There are severeal localization warning which should be remove before the release.

Especially if another language than English is used some warnings arise as already localized Strings are used as key to get the localization.

**Comment by koppor** [comment]
Could you provide an example? :innocent:

**Comment by matthiasgeiger** [comment]
Sure... 
```
WARNUNG: Warning: could not get translation for "JabRef Hilfe" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "Hilfe - Inhalt" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "Über JabRef" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "Eintrag bearbeiten" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "Tastatur-Fokus auf die Tabelle setzen" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "Datei speichern" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "Datei speichern unter ..." for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "Auswahl speichern unter ..." for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "Ausgewählte Einträge als reines BibTeX speichern …" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "Rückgängig" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "Wiederholen" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "Vor" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "Zurück" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "Löschen" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "Einträge markieren" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "Markierung aufheben" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "Relevanz ein-/ausschalten" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "Qualität ein-/ausschalten" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "Druckstatus umschalten" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "Weitersuchen (bei direkter Suche)" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "Suchen" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "Suchdialog ein-/ausblenden" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "Inhalt einer BibTeX-Datei an die aktuelle Datei anhängen" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "Präambel bearbeiten" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "Strings bearbeiten" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "Gruppenansicht ein-/ausblenden" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "Eintragsvorschau ein-/ausblenden" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "Gruppen markieren, die mindestens einen der selektierten Einträge enthalten" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "Gruppen markieren, die alle selektierten Einträge enthalten" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "BibTeX-Keys automatisch generieren" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "Schreibe XMP-Metadaten in die PDFs, die mit den ausgewählten Einträgen verlinkt sind." for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "Ordner öffnen" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "Datei öffnen" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "PDF oder PS öffnen" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "URL oder DOI öffnen" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "SPIRES-Eintrag öffnen" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "Zeitschriftentitel der ausgewählten Einträge abkürzen (MEDLINE-Abkürzung)" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "Zeitschriftentitel der ausgewählten Einträge abkürzen (ISO-Abkürzung)" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "Abkürzung der Zeitschriftentitel der ausgewählten Einträge aufheben" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "Mit externer SQL-Datenbank verbinden" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "Export in externe SQL-Datenbank" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "Einträge aufräumen" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "Einträge zusammenführen" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "Finde und entferne doppelte BibTeX-Keys" for locale de
Sep 10, 2015 9:38:13 AM net.sf.jabref.logic.l10n.Localization lang
WARNUNG: Warning: could not get translation for "Sucht nach nicht verlinkten PDF-Dateien im Dateisystem" for locale de
```

**Comment by mlep** [comment]
By default, JabRef is in English. What is strange here is that the warning displays a string in German and says it does not have the translation for it...

**Comment by matthiasgeiger** [comment]
That's exactly the problem here: Some strings are "double localized", i.e., somewhere in the code `Localization.lang(*english.key*)` is called and the result (a localized string) is used somewhere again in `Localization.lang(*localized.string*)` which rises this warnings containing the localized string. 
As the implementation returns simply the parameter if no localization is found for a key, this has no implications on the actual shown Strings in the GUI, however, the output is cluttered with this warnings.

**Comment by simonharrer** [comment]
Maybe this regex can help, as it finds all usage of Localization.lang with variables only `Localization\.lang\([^"]*\)`

And the variables could contain already translated strings.

**Comment by matthiasgeiger** [comment]
Most problems fixed - but activation of logging for missing menu items revealed some more problems.
I will investigate the remaining issues soon.

**Comment by matthiasgeiger** [comment]
Reverted as this might cause problems with the Python script, I got to know this afternoon :wink:

**Comment by matthiasgeiger** [comment]
Fixed with 23c64b5d899e8e04b56003bd0fdb45245a59b5c2, 250d1225ba6388815be7b3c9c0ce58cfb148b80c, 6edd76ab8bb7070a0904df65e197987f6203f8ce and 39cef04105422948a85747ec702c162a8c84b815.
